src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +52,170 @@ class MissingProjectMembership extends React.Component<Props, State> {
+    return [request, pending];
   }
 
-  renderJoinTeams(features: Set<string>) {
-    const teams = this.state.project?.teams ?? [];
+  handleChangeTeam = (team: string) => {
+    this.setState({team});
+  };
 
-    if (!teams.length) {
-      return (
-        <EmptyMessage>
-          {t(
-            'No teams have access to this project yet. Ask an admin to add your team to this project.'
-          )}
-        </EmptyMessage>
-      );
-    }
-
-    return teams.map((team: Team) => (
-      <p key={team.slug}>
-        #{team.slug}: {this.renderJoinTeam(team, features)}
-      </p>
-    ));
-  }
+  getPendingTeamOption = (team: string) => {
+    return {
+      value: team,
+      label: <DisabledLabel>{`#${team}`}</DisabledLabel>,
+    };
+  };
 
   render() {
     const {organization} = this.props;
+    const team = this.state.team;
+    const teams = this.state.project?.teams ?? [];
     const features = new Set(organization.features);
 
+    const teamAccess = [
+      {
+        label: t('Request Access'),
+        options: this.getTeamsForAccess()[0].map(request => ({
+          value: request,
+          label: `#${request}`,
+        })),
+      },
+      {
+        label: t('Pending Requests'),
+        options: this.getTeamsForAccess()[1].map(pending =>
+          this.getPendingTeamOption(pending)
+        ),
+      },
+    ];
+
     return (
-      <div className="container">
-        <StyledWell centered>
-          <StyledIconFlag size="xxl" />
-          <p>{t("You're not a member of this project.")}</p>
-          <p>{this.renderExplanation(features)}</p>
-          {this.renderJoinTeams(features)}
-        </StyledWell>
-      </div>
+      <StyledPanel>
+        {!teams.length ? (
+          <EmptyMessage icon={<IconFlag size="xl" />}>
+            {t(
+              'No teams have access to this project yet. Ask an admin to add your team to this project.'
+            )}
+          </EmptyMessage>
+        ) : (
+          <EmptyMessage
+            icon={<IconFlag size="xl" />}
+            title={t("You're not a member of this project.")}
+            description={t(
+              `You'll need to join a team with access before you can view this data.`
+            )}
+            action={
+              <StyledField>
+                <StyledSelectControl
+                  name="select"
+                  placeholder={t('Select a Team')}
+                  options={teamAccess}
+                  onChange={this.handleChangeTeam}
+                />
+                {team ? (
+                  this.renderJoinTeam(team, features)
+                ) : (
+                  <StyledButton disabled>{t('Select a Team')}</StyledButton>
+                )}
+              </StyledField>
+            }
+          />
+        )}
+      </StyledPanel>
     );
   }
 }
 
-const StyledWell = styled(Well)`
-  margin-top: ${space(2)};
+const StyledPanel = styled(Panel)`
+  margin: 15px 0 15px 0;
```

Comment:
nit: this can be shortened to `15px 0` - how does it look using `space(2)` (16px)?